MusicTaste/MusicTaste/APIManagers/SpotifyAPIManager.m

```diff
@@ -6,9 +6,166 @@
+                    completion(@{@"artists": artistDict[@"artists"], @"tracks":tracksDict[@"tracks"], @"albums": tracksDict[@"albums"], @"genres": artistDict[@"genres"], @"images": artistDict[@"images"]}, nil);
+                }
+            }];
+        }
+    }];
+}
+
+
+-(void) getSpotifyData:(NSString *)urlString completion:(void (^)(NSDictionary *, NSError*))completion{
+    NSURL *url = [NSURL URLWithString:urlString];
+    NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:url cachePolicy:NSURLRequestReloadIgnoringLocalCacheData timeoutInterval:10.0];
+    NSURLSession *session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration] delegate:nil delegateQueue:[NSOperationQueue mainQueue]];
+    [request setHTTPMethod:@"GET"];
+    [request addValue:[@"Bearer " stringByAppendingString:self.token] forHTTPHeaderField:@"Authorization"];
+    [request addValue:@"application/json" forHTTPHeaderField:@"Accept"];
+    [request addValue:@"application/json" forHTTPHeaderField:@"Content-Type"];
+    NSURLSessionDataTask *task = [session dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+           if (error != nil) {
+               completion(nil, error);
+           }
+           else {
+               NSDictionary *dataDictionary = [NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingMutableContainers error:nil];
+               completion(dataDictionary, nil);
+           }
+       }];
+    [task resume];
+}
+
+
+- (NSDictionary *) convertSpotifyArtists:(NSArray *)artists{
+    NSMutableSet *genreSet = [[NSMutableSet alloc] init];
+    NSMutableSet *artistSet = [[NSMutableSet alloc] init];
+    NSMutableArray *artistImages = [NSMutableArray new];
+    for (NSDictionary *artist in artists){
+        if  (artistImages.count < 10){
+            NSArray *artistAndImage = [NSArray arrayWithObjects: artist[@"name"],artist[@"images"][2][@"url"], nil];
+            [artistImages addObject:artistAndImage];
+        }
+        [genreSet addObjectsFromArray:artist[@"genres"]];
+        [artistSet addObject:artist[@"id"]];
+    }
+    return @{@"genres": genreSet, @"artists": artistSet, @"images": artistImages};
+}
+
+- (NSDictionary *) convertSpotifyTracks:(NSArray *)tracks{
+    NSMutableSet *trackSet = [[NSMutableSet alloc] init];
+    NSMutableSet *artistSet = [[NSMutableSet alloc] init];
+    NSMutableSet *albumSet = [[NSMutableSet alloc] init];
+    for (NSDictionary *track in tracks){
+        NSDictionary *albumItems = track[@"album"];
+        // adds album and artist of album to set
+        [albumSet addObject:albumItems[@"id"]];
+        for (NSDictionary *artist in albumItems[@"artists"]){
+            [artistSet addObject:artist[@"id"]];
+        }
+        // adds artists on song to set
+        NSArray *trackArtists = track[@"artists"];
+        if (trackArtists.count>1){
```

Comment:
Try to format your code, for readability `if (trackArtists.count > 1)`. XCode can do that for you automatically; I encourage you to find or assign a keyboard shortcut for that and use it regularly.